linux_os/guide/system/permissions/restrictions/enable_execshield_settings/sysctl_kernel_exec_shield/ansible/shared.yml

```
@@ -0,0 +1,39 @@
+# platform = multi_platform_all
+
+# What architecture are we on?
+- name: Set architecture for kernel exec-shield tasks
+  set_fact:
+    kexec_arch: "b{{ ansible_architecture | regex_replace('.*(\\d\\d$)','\\1') }}"
+
+- name: Ensure sysctl kernel.exec-shield is set to 1
+  sysctl:
+    name: "kernel.exec-shield"
+    value: "1"
+    state: present
+    reload: yes
+  when: kexec_arch == "b32"
+
+- name: Ensure existing of grub config
+  stat:
+    path: "{{ item.path }}"
+  register: stat_results
+  with_items:
+    - { path: /etc/grub2.cfg }
+    - { path: /etc/default/grub }
+  ignore_errors: True
+
+- name: Ensure noexec is not present in grub config files
+  replace:
+    path: "{{ item.item }}"
+    regexp: 'noexec.*'
+    replace: ''
+  with_items:
+    - { path: /etc/grub2.cfg }
+    - { path: /etc/default/grub }
+  with_items: "{{ stat_results.results }}"
+  when: ( kexec_arch == "b64" ) and ( item.stat.exists == False )
```

Don’t compare to literal True/False

```suggestion
  when: ( kexec_arch == "b64" ) and ( not item.stat.exists )
```

---

Thanks for the suggestion @jan-cerny